rediscluster/pipeline.py

```diff
@@ -193,13 +198,19 @@ def send_cluster_commands(self, stack, raise_on_error=True, allow_redirections=T
         # a mismatched result. (not just theoretical, I saw this happen on production x.x).
         for n in nodes.values():
             self.connection_pool.release(n.connection)
+            if n.connection in self.connection_pool._available_connections.get(node_name, []):
+                self.connection_pool._available_connections[node_name].remove(n.connection)
+
+            if self.connection_pool._created_connections_per_node.get(node_name, 0):
+                self.connection_pool._created_connections_per_node[node_name] -= 1
```

Comment:
I am not sure, but should a connection be discarded directly? The connection pool then seems to be meaningless